#320 Fix Sortable Tables

Keegan-lee opened the pull request:
#### What's this PR do?
- Revamped sorting mechanism to fix the issue
- Fixes the broken sortability

#### Where should the reviewer start?
- SortableTable.vue
#### How should this be manually tested?
Make sure that the tables can be toggled programmatically. Toggle a sortable table on/off and make sure that the tables remain sortable.

#### Any background context you want to provide?
- This is a fix for ticket #306 
#### What are the relevant tickets?
- #306, #320 
#### Screenshots (if appropriate)
#### URL
[http://staging-ui.trusty.apasia.tech:8080/branch-320SortabilityIsBroken](http://staging-ui.trusty.apasia.tech:8080/branch-320SortabilityIsBroken)
#### Questions:

Commit:
#320 Fix Sortable Tables
- Revamped sorting mechanism to fix the issue

roma219 commented on src/components/SortableTable/SortableTable.vue:
```diff
@@ -108,7 +108,8 @@ export default {
   methods: {
-    toggleSort(field, header) {
+    toggleSort(field) {
+      let header = this.headers.find(h => h.field === field) || {}
```
I think this kind of check should be performed inside `SortableHeaderItem` - the event should not be emitted if sorting is disabled

Keegan-lee replied:
I agree! I figured out how to do this elsewhere.

roma219 reviewed (changes requested):
Improvement suggested

Commit:
#320 Sortable Table Improvement
- Moved the disabling of the sortable header into the SortableHeaderItem component.

roma219 commented on src/components/SortableHeaderItem/SortableHeaderItem.vue:
```diff
@@ -94,6 +94,11 @@ export default {
       type: String,
       default: ''
+    },
+    disableSort: {
```
I don't think you need this prop as you already have access to `header` object as `item`
